lib/ecto/changeset.ex

```diff
@@ -1419,6 +1422,7 @@ defmodule Ecto.Changeset do
       |> unique_constraint(:email)
 
   """
+  @spec unique_constraint(t, atom, options) :: t
```

Comment:
@michalmuskala I added those specs according to the usage of `options` in the functions. Those with the type `options` rely on Access, so maps would also work. Where it says `Keyword.t`, the function uses `Keyword` for fetching the options.